LE Audio: BAP shell improvements

Thalley opened the pull request:
This PR improves the BAP shell by

- Making presets constant, and moving stream-specific configuration into a new unicast_stream struct
- Improve metadata handling
- Refactor handling of QoS
- Refactor handling of default presets

Thalley committed:
Bluetooth: Audio: Shell: Rename streams to unicast_streams

The streams array only contained unicast streams, and
the name should reflect that.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Shell: Make BAP shell presets constant

Make the BAP shell presets constant. This reflects more
what they represent, and make it not only possible, but also
easier, to modify the individual streams QoS and Codec
configurations, without affecting other streams.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Shell: Replace hacky metadata handling

The hacky handle_metadata_update that copied the metadata to
a variable, since stream->codec->meta could not be modified,
has been replaced with a proper solution now that
stream->codec->meta can be modified.

It still only supports settings the streaming context, but
it is much easier to expand now.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Shell: Refactor set_preset

The function is now called get_named_preset and only
returns a named preset. The QoS parameter handling
has been moved to a new function, cmd_stream_qos,
which can be used to set specific QoS for a stream.

The get_named_preset does also no longer set the default
preset, and cmd_preset must be used for that now.

This means that a function like cmd_config no longer
has any side effects, thus making it easier to set up
multiple streams with different presets.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

larsgk reviewed (changes requested):
Please update docs too :)

Thalley committed:
Bluetooth: Audio: Shell: Update BAP shell documentation

Update the BAP shell documentation to be more up to date.
There are still several BAP commands that do not have
examples of further documentation, but that is still
a TODO.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Shell: Add location to cmd_config

Add support for setting the location value
(BT_CODEC_CONFIG_LC3_CHAN_ALLOC) in the codec configuration.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley commented:
Fixed segmentation fault and added additional commit to handle setting stream location as part of `cmd_config`
